### PR TITLE
Add deprecation guide

### PR DESCRIPTION
**Pull request by Faradey27**
### What changed

...

### Why it changed

...

---

- [x] Change is tested
- [x] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

**Comment by Faradey27** [comment]
@shlomitcI will be really thankful for your review and text recommendations

File: docs/DEPRECATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# How to deprecate?
+
+## Deprecating components
+1. Move your component to `wix-style-react/src/Deprecated` folder
+2. Update all internal imports of your component
```
Comment (shlomitc) [inline docs/DEPRECATION_GUIDE.md]: this is trivial

File: docs/DEPRECATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+3. Check that all tests are passing
```
Comment (shlomitc) [inline docs/DEPRECATION_GUIDE.md]: if component moves as it is, should be trivial

File: docs/DEPRECATION_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+4. Use  deprecationLog to warn the user that he using deprecated component and explain him his next steps
```
Comment (shlomitc) [inline docs/DEPRECATION_GUIDE.md]: this should be moved to the previous folder where it sits and not inside the component

File: docs/DEPRECATION_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  class SomeDeprecatedComponent extends React.Component {
+    constructor(props) {
+      super(props);
+      deprecationLog('"SomeDeprecatedComponent" was deprecated, please use "AnotherComponent" instead');
+    }
+    ...
+  }
+```
+5. Do not forgot to add test, that check warn message
```
Comment (shlomitc) [inline docs/DEPRECATION_GUIDE.md]: I don't think it's necessary, it takes time and it's not necessarily something we want to maintain

File: docs/DEPRECATION_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+# How to deprecate?
+
+## Deprecating components
+1. Move your component to `wix-style-react/src/Deprecated` folder
```
[the remark this answers is not in the document]
Comment (Faradey27) [reply docs/DEPRECATION_GUIDE.md]: @shlomitc no, http://prntscr.com/jxy34i upper case

File: docs/DEPRECATION_GUIDE.md (hunk 1 above)
Comment (Faradey27) [reply docs/DEPRECATION_GUIDE.md]: people usually super lazy and forgot everything)

File: docs/DEPRECATION_GUIDE.md (hunk 4 above)
Comment (Faradey27) [reply docs/DEPRECATION_GUIDE.md]: ok

**Comment by Faradey27** [comment]
@shlomitc updated

**Pull request merged**